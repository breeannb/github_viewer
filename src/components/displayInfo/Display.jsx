import React from 'react';
import PropTypes from 'prop-types';

const Display = ({ name, followers, following, html_url, repos }) => (
  <>
  
    <p>Username: {name}</p>
    <ul>
      <li>Followers: {followers}</li>
      <li>Following: {following}</li>
      <li>
        <a href={html_url}>Profile Link</a>
      </li>
    </ul>

    <p> Repositories: </p>
    <ul>
      <li>
        {
          repos.map(repo => 
            <li key={repo.id}>
              <a href={repo.html_url}>{repo.name}</a>
            </li>)
        }
      </li>
    </ul>
    
  </>
);

Display.propTypes = {
  name: PropTypes.string.isRequired,
  followers: PropTypes.number.isRequired,
  following: PropTypes.number.isRequired,
  html_url: PropTypes.string.isRequired, 
  repos: PropTypes.arrayOf(PropTypes.object).isRequired
};

export default Display;
