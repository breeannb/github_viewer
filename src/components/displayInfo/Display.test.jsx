import React from 'react';
import { shallow } from 'enzyme';
import Display from './Display';

describe('App component', () => {
  it('renders App', () => {
    const wrapper = shallow(<Display username="Breeann" followers={4} following={200} html_url={'https://github.com/breeannb'} repos={[{ name: 'Breeann', html_url: 'https://github.com/breeannb' }]}/>);
    expect(wrapper).toMatchSnapshot();
  });
});
