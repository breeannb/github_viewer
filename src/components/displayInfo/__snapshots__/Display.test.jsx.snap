// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<Fragment>
  <p>
    Username: 
  </p>
  <ul>
    <li>
      Followers: 
      4
    </li>
    <li>
      Following: 
      200
    </li>
    <li>
      <a
        href="https://github.com/breeannb"
      >
        Profile Link
      </a>
    </li>
  </ul>
  <p>
     Repositories: 
  </p>
  <ul>
    <li>
      <li>
        <a
          href="https://github.com/breeannb"
        >
          Breeann
        </a>
      </li>
    </li>
  </ul>
</Fragment>
`;
